src/index.js

```diff
@@ -114,46 +137,50 @@ async function fetchWithNetworkEvents(el: Element, url: string, options: Request
 }
 
 async function check(autoCheckElement: AutoCheckElement) {
-  const src = autoCheckElement.src
-  if (!src) {
-    throw new Error('missing src')
-  }
-  const csrf = autoCheckElement.csrf
-  if (!csrf) {
-    throw new Error('missing csrf')
-  }
   const input = autoCheckElement.input
-  if (!input) return
+  if (!input) {
+    return
+  }
 
+  const src = autoCheckElement.src
+  const csrf = autoCheckElement.csrf
   const state = states.get(autoCheckElement)
-  if (!state) return
+
+  // If some attributes are missing we want to exit early and make sure that the element is valid.
+  if (!src || !csrf || !state) {
+    if (autoCheckElement.required) {
+      input.setCustomValidity('')
```

Comment:
Looking at it now with `-complete` event, I feel like it sholuld error instead? at least when `required`.

Reply:
That makes sense to me. Before we threw a error and halted execution but emitting an error event here instead might be the way to go. It lets users clean up after setting up in the `auto-check-input` event handler.

Reply:
Error being setting the input to invalid state? Or this should be throwing a error?

Reply:
IMO either emitting `auto-check-error` + invalidate the input, or throw, like it previously did.

Reply:
I took [@dgraham's suggestion](https://github.com/github/auto-check-element/pull/35#discussion_r340834921) here and kept it as is.

Reply:
I think that suggestion should apply to not sending `auto-check-start` as well.